api/src/Modules/staff/ModuleStaff.php

```diff
@@ -73,39 +73,49 @@ public function customzieStaffRBAC($rbac, $database)
             $dept = $value['DepartmentID'];
 
             if ($parentDept != $dept) {
-                $targetRole = "$parentDept.".end(array_keys($positions));
-                try {
-                    $rbac->grantPermission($targetRole, "api.put.staff.$dept.1");
-                    $rbac->grantPermission($targetRole, "api.delete.staff.$dept.1");
-                } catch (Exception\InvalidArgumentException $e) {
-                    error_log($e);
+                // Grants for Dept Head - They can add members to their department
+                $deptHeadPos = reset(array_keys($positions));
+                $deptHeadRole = "$dept.$deptHeadPos";
+                self::grantPermission($rbac, $deptHeadRole, "api.post.staff.$dept");
+
+                // Grants for Division Staff - They can add to any department within their division, and they can edit any department members within their division.
+                foreach ($positions as $directorPosId => $directorPos) {
+                    $targetRole = "$parentDept.$directorPosId";
+                    self::grantPermission($rbac, $targetRole, "api.post.staff.$dept");
+
+                    foreach ($positions as $deptPosId => $deptPos) {
+                        self::grantPermission($rbac, $targetRole, "api.put.staff.$dept.$deptPosId");
+                        self::grantPermission($rbac, $targetRole, "api.delete.staff.$dept.$deptPosId");
+                    }
                 }
```

Comment:
Yes, it's because I realized we were called `$rbac->grantPermission` multiple times and wanted to make this DRY. `grantPermission` on this class helps us encapsulate the error handling logic and we don't have to repeat it in multiple places.